app/assets/javascripts/usage_charts.js

```diff
@@ -4,27 +4,23 @@ $(document).ready(function() {
   //called by event handlers that need to update the graph
   //they should have already made any changes to the form we're using, so this just updates
   //the explanation and then triggers the data load
-  function updateChart(el) {
+  function updateChart(chartDiv) {
 
-    var supply = $("#supply").val();
-    var period = $("#period").val();
-    var config = $("#configuration").data('configuration');
-    var descriptions = $("#descriptions").data('descriptions');
-
-    var chartContainer = $('.usage-chart').first();
+    var descriptions = $(chartDiv).find("input[name='descriptions']").data('descriptions');
+    var chartContainer = $(chartDiv).find('.usage-chart').first();
     var chartConfig = chartContainer.data('chart-config');
 
-    var meter = $("#meter").val();
+    var meter = $(chartDiv).find("select[name='meter']").val();
     if (meter && meter != 'all') {
       chartConfig.mpan_mprn = meter;
     }
 
-    var series_breakdown = $("#series_breakdown").val();
+    var series_breakdown = $(chartDiv).find("input[name='series_breakdown']").val();
```

Comment:
should `series_breakdown` be `seriesBreakdown`?  (not that i have a problem with using snake case in js but i guess we need to stick to conventions)